src/requestHandler.ts

```diff
@@ -383,8 +373,11 @@ export default class RequestHandler {
       return;
     }
 
+    await createDirNotExist(filepath)
+    await this.app.vault.adapter.write(filepath, req.body);
```

Comment:
I might be misreading this, but it kind of looks like were changing the implementation of `vaultPost` here from:

1. Checking if there's an existing file and reading its contents if necessary
2. Writing the file contents from (1) and then appending the incoming request body

to:

1. Creating the directory parent directories if they don't already exist.
2. Writing the request body to the file.
3. Checking if there's an existing file and reading its contents if necessary.
4. Writing the the file contents from (3) and appending the incoming request body.

I _think_ the right choice here is to just remove line 377 ((2) above), but I might be misunderstanding your intentions here.

Reply:
You are right. MB XD